### PR TITLE
Taskfiles: Rearrange task attributes to match latest style guide.

### PR DESCRIPTION
**Pull request by Henry8192**
# Reference
We have a guideline at https://docs.yscope.com/dev-guide/contrib-guides-taskfiles.html#ordering-of-task-attributes for organizing task attributes in Taskfile.

# Description
`<clp-root>/Taskfile.yml` and `<clp-root>/lint-task.yml` were not formatted properly. 
1. Updated task attributes orders in both files according to the guidelines.

# Validation performed
Run checker `lint-taskfile.py` and observed no format error.
[lint-taskfile.zip](https://github.com/user-attachments/files/16199333/lint-taskfile.zip)

### CODE REVIEW

**Comment by junhaoliao** [comment]
It seem the attributes in task `webui` is not ordered as specified in the guideline. Did the linter report any error on that?

**Comment by Henry8192** [comment]
> It seem the attributes in task `webui` is not ordered as specified in the guideline. Did the linter report any error on that?

Strange. Only two errors were generated: core and package.

**Comment by junhaoliao** [comment]
> > It seem the attributes in task `webui` is not ordered as specified in the guideline. Did the linter report any error on that?
> 
> Strange. Only two errors were generated: core and package.

Did you see any error when running the script? I see plenty here because some `attrs` isn't of type `dict`.

**Comment by junhaoliao** [comment]
Here's the script fixed: 
[lint-taskfile.zip](https://github.com/user-attachments/files/16199705/lint-taskfile.zip)

Please take a look into these errors:
```
Taskfile.yml: Task 'log-viewer-webui' has unsorted attributes.
	Expected order: ['vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
	Actual order  : ['deps', 'dir', 'vars', 'cmds', 'sources', 'generates']
Taskfile.yml: Task 'webui' has unsorted attributes.
	Expected order: ['vars', 'sources', 'dir', 'platforms', 'generates', 'deps', 'cmds']
	Actual order  : ['deps', 'dir', 'platforms', 'vars', 'cmds', 'sources', 'generates']
Taskfile.yml: Task 'core-submodules' has unsorted attributes.
	Expected order: ['internal', 'vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
	Actual order  : ['internal', 'dir', 'vars', 'deps', 'cmds', 'sources', 'generates']
Taskfile.yml: Task 'download-and-extract-tar' has unsorted attributes.
	Expected order: ['internal', 'label', 'vars', 'requires', 'sources', 'generates', 'deps', 'cmds']
	Actual order  : ['internal', 'requires', 'label', 'vars', 'deps', 'cmds', 'sources', 'generates']
Taskfile.yml: Task 'log-viewer-webui-node-modules' has unsorted attributes.
	Expected order: ['internal', 'vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
	Actual order  : ['internal', 'vars', 'dir', 'deps', 'cmds', 'sources', 'generates']
Taskfile.yml: Task 'meteor' has unsorted attributes.
	Expected order: ['vars', 'run', 'preconditions', 'cmds']
	Actual order  : ['run', 'preconditions', 'vars', 'cmds']
Taskfile.yml: Task 'nodejs' has unsorted attributes.
	Expected order: ['internal', 'vars', 'requires', 'deps', 'cmds']
	Actual order  : ['internal', 'deps', 'requires', 'vars', 'cmds']
Taskfile.yml: Task 'package-tar' has unsorted attributes.
	Expected order: ['internal', 'vars', 'requires', 'sources', 'dir', 'generates', 'deps', 'cmds']
	Actual order  : ['internal', 'requires', 'vars', 'dir', 'deps', 'cmds', 'sources', 'generates']
Taskfile.yml: Task 'package-venv' has unsorted attributes.
	Expected order: ['internal', 'vars', 'sources', 'generates', 'deps', 'cmds']
	Actual order  : ['internal', 'vars', 'deps', 'cmds', 'sources', 'generates']
Taskfile.yml: Task 'python-component' has unsorted attributes.
	Expected order: ['internal', 'label', 'vars', 'requires', 'sources', 'dir', 'generates', 'deps', 'cmds']
	Actual order  : ['internal', 'requires', 'label', 'deps', 'vars', 'dir', 'cmds', 'sources', 'generates']
Taskfile.yml: Task 'webui-node-modules' has unsorted attributes.
	Expected order: ['internal', 'vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
	Actual order  : ['internal', 'vars', 'dir', 'deps', 'cmds', 'sources', 'generates']
Taskfile.yml: Task 'component-venv' has unsorted attributes.
	Expected order: ['internal', 'label', 'vars', 'requires', 'sources', 'dir', 'generates', 'deps', 'cmds']
	Actual order  : ['internal', 'requires', 'label', 'dir', 'vars', 'deps', 'cmds', 'sources', 'generates']
Taskfile.yml: Task 'clean-python-component' has unsorted attributes.
	Expected order: ['internal', 'label', 'requires', 'dir', 'cmds']
	Actual order  : ['internal', 'requires', 'label', 'dir', 'cmds']
Taskfile.yml: Task 'init' has unsorted attributes.
	Expected order: ['internal', 'silent', 'run', 'cmd']
	Actual order  : ['internal', 'run', 'silent', 'cmd']
```

Also, as we discussed offline, please check https://github.com/y-scope/clp/blob/main/lint-tasks.yml as well.
```
lint-tasks.yml: Task 'cpp-check' has unsorted attributes.
	Expected order: ['sources', 'dir', 'cmds']
	Actual order  : ['dir', 'cmds', 'sources']
lint-tasks.yml: Task 'cpp-fix' has unsorted attributes.
	Expected order: ['sources', 'dir', 'cmds']
	Actual order  : ['dir', 'cmds', 'sources']
lint-tasks.yml: Task 'js-check' has unsorted attributes.
	Expected order: ['sources', 'cmds']
	Actual order  : ['cmds', 'sources']
lint-tasks.yml: Task 'js-fix' has unsorted attributes.
	Expected order: ['sources', 'cmds']
	Actual order  : ['cmds', 'sources']
lint-tasks.yml: Task 'cpp' has unsorted attributes.
	Expected order: ['internal', 'requires', 'dir', 'deps', 'cmds']
	Actual order  : ['internal', 'requires', 'deps', 'dir', 'cmds']
lint-tasks.yml: Task 'linter-node-modules' has unsorted attributes.
	Expected order: ['internal', 'vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
	Actual order  : ['internal', 'deps', 'dir', 'vars', 'cmds', 'sources', 'generates']
lint-tasks.yml: Task 'venv' has unsorted attributes.
	Expected order: ['internal', 'vars', 'sources', 'generates', 'deps', 'cmds']
	Actual order  : ['internal', 'vars', 'deps', 'cmds', 'sources', 'generates']
```

**Comment by junhaoliao** [comment]
@Henry8192 When you're done with this PR, please also submit another PR to update task attributes here:
https://github.com/y-scope/yscope-docs/blob/1c9227f7608854b6eab5e09410da8c1f666eff22/docs/dev-guide/contrib-guides-taskfiles.md?plain=1#L212
1. `platform` -> `platforms`.
2. More might be modified depending on some offline discussion results.

**Comment by Henry8192** [comment]
> > > It seem the attributes in task `webui` is not ordered as specified in the guideline. Did the linter report any error on that?
> 
> > 
> 
> > Strange. Only two errors were generated: core and package.
> 
> 
> 
> Did you see any error when running the script? I see plenty here because some `attrs` isn't of type `dict`.

Yeah I thought it was my script config error

**Comment by Henry8192** [comment]
> Here's the script fixed: 
> 
> [lint-tasks.zip](https://github.com/user-attachments/files/16199758/lint-tasks.zip)
> 
> 
> 
> Please take a look into these errors:
> 
> ```
> 
> Taskfile.yml: Task 'log-viewer-webui' has unsorted attributes.
> 
> 	Expected order: ['vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
> 
> 	Actual order  : ['deps', 'dir', 'vars', 'cmds', 'sources', 'generates']
> 
> Taskfile.yml: Task 'webui' has unsorted attributes.
> 
> 	Expected order: ['vars', 'sources', 'dir', 'platforms', 'generates', 'deps', 'cmds']
> 
> 	Actual order  : ['deps', 'dir', 'platforms', 'vars', 'cmds', 'sources', 'generates']
> 
> Taskfile.yml: Task 'core-submodules' has unsorted attributes.
> 
> 	Expected order: ['internal', 'vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
> 
> 	Actual order  : ['internal', 'dir', 'vars', 'deps', 'cmds', 'sources', 'generates']
> 
> Taskfile.yml: Task 'download-and-extract-tar' has unsorted attributes.
> 
> 	Expected order: ['internal', 'label', 'vars', 'requires', 'sources', 'generates', 'deps', 'cmds']
> 
> 	Actual order  : ['internal', 'requires', 'label', 'vars', 'deps', 'cmds', 'sources', 'generates']
> 
> Taskfile.yml: Task 'log-viewer-webui-node-modules' has unsorted attributes.
> 
> 	Expected order: ['internal', 'vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
> 
> 	Actual order  : ['internal', 'vars', 'dir', 'deps', 'cmds', 'sources', 'generates']
> 
> Taskfile.yml: Task 'meteor' has unsorted attributes.
> 
> 	Expected order: ['vars', 'run', 'preconditions', 'cmds']
> 
> 	Actual order  : ['run', 'preconditions', 'vars', 'cmds']
> 
> Taskfile.yml: Task 'nodejs' has unsorted attributes.
> 
> 	Expected order: ['internal', 'vars', 'requires', 'deps', 'cmds']
> 
> 	Actual order  : ['internal', 'deps', 'requires', 'vars', 'cmds']
> 
> Taskfile.yml: Task 'package-tar' has unsorted attributes.
> 
> 	Expected order: ['internal', 'vars', 'requires', 'sources', 'dir', 'generates', 'deps', 'cmds']
> 
> 	Actual order  : ['internal', 'requires', 'vars', 'dir', 'deps', 'cmds', 'sources', 'generates']
> 
> Taskfile.yml: Task 'package-venv' has unsorted attributes.
> 
> 	Expected order: ['internal', 'vars', 'sources', 'generates', 'deps', 'cmds']
> 
> 	Actual order  : ['internal', 'vars', 'deps', 'cmds', 'sources', 'generates']
> 
> Taskfile.yml: Task 'python-component' has unsorted attributes.
> 
> 	Expected order: ['internal', 'label', 'vars', 'requires', 'sources', 'dir', 'generates', 'deps', 'cmds']
> 
> 	Actual order  : ['internal', 'requires', 'label', 'deps', 'vars', 'dir', 'cmds', 'sources', 'generates']
> 
> Taskfile.yml: Task 'webui-node-modules' has unsorted attributes.
> 
> 	Expected order: ['internal', 'vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
> 
> 	Actual order  : ['internal', 'vars', 'dir', 'deps', 'cmds', 'sources', 'generates']
> 
> Taskfile.yml: Task 'component-venv' has unsorted attributes.
> 
> 	Expected order: ['internal', 'label', 'vars', 'requires', 'sources', 'dir', 'generates', 'deps', 'cmds']
> 
> 	Actual order  : ['internal', 'requires', 'label', 'dir', 'vars', 'deps', 'cmds', 'sources', 'generates']
> 
> Taskfile.yml: Task 'clean-python-component' has unsorted attributes.
> 
> 	Expected order: ['internal', 'label', 'requires', 'dir', 'cmds']
> 
> 	Actual order  : ['internal', 'requires', 'label', 'dir', 'cmds']
> 
> Taskfile.yml: Task 'init' has unsorted attributes.
> 
> 	Expected order: ['internal', 'silent', 'run', 'cmd']
> 
> 	Actual order  : ['internal', 'run', 'silent', 'cmd']
> 
> ```
> 
> 
> 
> Also, as we discussed offline, please check https://github.com/y-scope/clp/blob/main/lint-tasks.yml as well.
> 
> ```
> 
> lint-tasks.yml: Task 'cpp-check' has unsorted attributes.
> 
> 	Expected order: ['sources', 'dir', 'cmds']
> 
> 	Actual order  : ['dir', 'cmds', 'sources']
> 
> lint-tasks.yml: Task 'cpp-fix' has unsorted attributes.
> 
> 	Expected order: ['sources', 'dir', 'cmds']
> 
> 	Actual order  : ['dir', 'cmds', 'sources']
> 
> lint-tasks.yml: Task 'js-check' has unsorted attributes.
> 
> 	Expected order: ['sources', 'cmds']
> 
> 	Actual order  : ['cmds', 'sources']
> 
> lint-tasks.yml: Task 'js-fix' has unsorted attributes.
> 
> 	Expected order: ['sources', 'cmds']
> 
> 	Actual order  : ['cmds', 'sources']
> 
> lint-tasks.yml: Task 'cpp' has unsorted attributes.
> 
> 	Expected order: ['internal', 'requires', 'dir', 'deps', 'cmds']
> 
> 	Actual order  : ['internal', 'requires', 'deps', 'dir', 'cmds']
> 
> lint-tasks.yml: Task 'linter-node-modules' has unsorted attributes.
> 
> 	Expected order: ['internal', 'vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
> 
> 	Actual order  : ['internal', 'deps', 'dir', 'vars', 'cmds', 'sources', 'generates']
> 
> lint-tasks.yml: Task 'venv' has unsorted attributes.
> 
> 	Expected order: ['internal', 'vars', 'sources', 'generates', 'deps', 'cmds']
> 
> 	Actual order  : ['internal', 'vars', 'deps', 'cmds', 'sources', 'generates']
> 
> ```

Nope I didn't see those errors...

**Comment by junhaoliao** [comment]
> > Here's the script fixed: 
> > 
> > [lint-tasks.zip](https://github.com/user-attachments/files/16199758/lint-tasks.zip)
> > 
> > 
> > 
> > Please take a look into these errors:
> > 
> > ```
> > 
> > Taskfile.yml: Task 'log-viewer-webui' has unsorted attributes.
> > 
> > 	Expected order: ['vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
> > 
> > 	Actual order  : ['deps', 'dir', 'vars', 'cmds', 'sources', 'generates']
> > 
> > Taskfile.yml: Task 'webui' has unsorted attributes.
> > 
> > 	Expected order: ['vars', 'sources', 'dir', 'platforms', 'generates', 'deps', 'cmds']
> > 
> > 	Actual order  : ['deps', 'dir', 'platforms', 'vars', 'cmds', 'sources', 'generates']
> > 
> > Taskfile.yml: Task 'core-submodules' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
> > 
> > 	Actual order  : ['internal', 'dir', 'vars', 'deps', 'cmds', 'sources', 'generates']
> > 
> > Taskfile.yml: Task 'download-and-extract-tar' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'label', 'vars', 'requires', 'sources', 'generates', 'deps', 'cmds']
> > 
> > 	Actual order  : ['internal', 'requires', 'label', 'vars', 'deps', 'cmds', 'sources', 'generates']
> > 
> > Taskfile.yml: Task 'log-viewer-webui-node-modules' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
> > 
> > 	Actual order  : ['internal', 'vars', 'dir', 'deps', 'cmds', 'sources', 'generates']
> > 
> > Taskfile.yml: Task 'meteor' has unsorted attributes.
> > 
> > 	Expected order: ['vars', 'run', 'preconditions', 'cmds']
> > 
> > 	Actual order  : ['run', 'preconditions', 'vars', 'cmds']
> > 
> > Taskfile.yml: Task 'nodejs' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'vars', 'requires', 'deps', 'cmds']
> > 
> > 	Actual order  : ['internal', 'deps', 'requires', 'vars', 'cmds']
> > 
> > Taskfile.yml: Task 'package-tar' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'vars', 'requires', 'sources', 'dir', 'generates', 'deps', 'cmds']
> > 
> > 	Actual order  : ['internal', 'requires', 'vars', 'dir', 'deps', 'cmds', 'sources', 'generates']
> > 
> > Taskfile.yml: Task 'package-venv' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'vars', 'sources', 'generates', 'deps', 'cmds']
> > 
> > 	Actual order  : ['internal', 'vars', 'deps', 'cmds', 'sources', 'generates']
> > 
> > Taskfile.yml: Task 'python-component' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'label', 'vars', 'requires', 'sources', 'dir', 'generates', 'deps', 'cmds']
> > 
> > 	Actual order  : ['internal', 'requires', 'label', 'deps', 'vars', 'dir', 'cmds', 'sources', 'generates']
> > 
> > Taskfile.yml: Task 'webui-node-modules' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
> > 
> > 	Actual order  : ['internal', 'vars', 'dir', 'deps', 'cmds', 'sources', 'generates']
> > 
> > Taskfile.yml: Task 'component-venv' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'label', 'vars', 'requires', 'sources', 'dir', 'generates', 'deps', 'cmds']
> > 
> > 	Actual order  : ['internal', 'requires', 'label', 'dir', 'vars', 'deps', 'cmds', 'sources', 'generates']
> > 
> > Taskfile.yml: Task 'clean-python-component' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'label', 'requires', 'dir', 'cmds']
> > 
> > 	Actual order  : ['internal', 'requires', 'label', 'dir', 'cmds']
> > 
> > Taskfile.yml: Task 'init' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'silent', 'run', 'cmd']
> > 
> > 	Actual order  : ['internal', 'run', 'silent', 'cmd']
> > 
> > ```
> > 
> > 
> > 
> > Also, as we discussed offline, please check https://github.com/y-scope/clp/blob/main/lint-tasks.yml as well.
> > 
> > ```
> > 
> > lint-tasks.yml: Task 'cpp-check' has unsorted attributes.
> > 
> > 	Expected order: ['sources', 'dir', 'cmds']
> > 
> > 	Actual order  : ['dir', 'cmds', 'sources']
> > 
> > lint-tasks.yml: Task 'cpp-fix' has unsorted attributes.
> > 
> > 	Expected order: ['sources', 'dir', 'cmds']
> > 
> > 	Actual order  : ['dir', 'cmds', 'sources']
> > 
> > lint-tasks.yml: Task 'js-check' has unsorted attributes.
> > 
> > 	Expected order: ['sources', 'cmds']
> > 
> > 	Actual order  : ['cmds', 'sources']
> > 
> > lint-tasks.yml: Task 'js-fix' has unsorted attributes.
> > 
> > 	Expected order: ['sources', 'cmds']
> > 
> > 	Actual order  : ['cmds', 'sources']
> > 
> > lint-tasks.yml: Task 'cpp' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'requires', 'dir', 'deps', 'cmds']
> > 
> > 	Actual order  : ['internal', 'requires', 'deps', 'dir', 'cmds']
> > 
> > lint-tasks.yml: Task 'linter-node-modules' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'vars', 'sources', 'dir', 'generates', 'deps', 'cmds']
> > 
> > 	Actual order  : ['internal', 'deps', 'dir', 'vars', 'cmds', 'sources', 'generates']
> > 
> > lint-tasks.yml: Task 'venv' has unsorted attributes.
> > 
> > 	Expected order: ['internal', 'vars', 'sources', 'generates', 'deps', 'cmds']
> > 
> > 	Actual order  : ['internal', 'vars', 'deps', 'cmds', 'sources', 'generates']
> > 
> > ```
> 
> Nope I didn't see those errors...

Previously there was an exception (printed to stderr) when running the old script with the Taskfile. Maybe somehow bytes written to stderr was swallowed.

File: Taskfile.yml [hunk 1]
```diff
@@ -597,27 +603,28 @@ tasks:
...
+      - "pyproject.toml"
+    dir: "components/{{.COMPONENT}}"
+    generates: ["{{.CHECKSUM_FILE}}"]
     deps:
       - "init"
       - task: "utils:validate-checksum"
         vars:
           CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
           DATA_DIR: "{{.OUTPUT_DIR}}"
+
```
Comment (junhaoliao) [inline Taskfile.yml:627]:
```suggestion
```

File: lint-tasks.yml [hunk 2]
```diff
@@ -210,9 +215,4 @@ tasks:
       - task: ":utils:compute-checksum"
         vars:
           DATA_DIR: "{{.OUTPUT_DIR}}"
-          OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
-    sources:
-      - "{{.ROOT_DIR}}/Taskfile.yml"
-      - "{{.TASKFILE}}"
-      - "lint-requirements.txt"
-    generates: ["{{.CHECKSUM_FILE}}"]
+          OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
```
Comment (junhaoliao) [inline lint-tasks.yml:218]:
```suggestion
          OUTPUT_FILE: "{{.CHECKSUM_FILE}}"

```

**Review by junhaoliao — APPROVED** [review]
LGTM

**Review by kirkrodrigues — APPROVED** [review]
For the PR title, how about:

Taskfiles: Rearrange task attributes to match latest style guide.

The current title implies there's an automated linter which might be a bit misleading.

**Pull request merged**